test/integration/ethereum/flashMintLeveraged.spec.ts

```diff
@@ -0,0 +1,466 @@
+                  expect(quotedInputAmount).to.lt(preciseMul(inputSpent, ether(1.01)));
+                });
+              },
+            );
+
+            describe(
+              inputTokenName == "ETH" ? "redeemExactSetForETH" : "#redeemExactSetForERC20",
+              () => {
+                let swapDataCollateralToDebt: SwapData;
+                let swapDataOutputToken: SwapData;
+
+                let outputToken: StandardTokenMock | IWETH;
+
+                let subjectSetToken: Address;
+                let subjectMinAmountOut: BigNumber;
+                let subjectOutputToken: Address;
+
+                async function subject() {
+                  if (inputTokenName == "ETH") {
+                    return flashMintLeveraged.redeemExactSetForETH(
+                      subjectSetToken,
+                      subjectSetAmount,
+                      subjectMinAmountOut,
+                      swapDataCollateralToDebt,
+                      swapDataOutputToken
+                    );
+                  }
+                  return flashMintLeveraged.redeemExactSetForERC20(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    subjectOutputToken,
+                    subjectMinAmountOut,
+                    swapDataCollateralToDebt,
+                    swapDataOutputToken
+                );
+                }
+
+                async function subjectQuote(): Promise<BigNumber> {
+                  return flashMintLeveraged.callStatic.getRedeemExactSet(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    swapDataCollateralToDebt, swapDataOutputToken
+                  );
+                }
+
+                beforeEach(async () => {
+                  swapDataCollateralToDebt = {
+                    path: [collateralTokenAddress, addresses.tokens.weth],
+                    fees: [500],
+                    pool: ADDRESS_ZERO,
+                    exchange: Exchange.UniV3,
+                  };
+
+                  if (inputTokenName == "collateralToken") {
+                    outputToken = rEth;
+                    swapDataOutputToken = {
+                      path: [],
+                      fees: [],
+                      pool: ADDRESS_ZERO,
+                      exchange: Exchange.None,
+                    };
+                  } else {
+                    swapDataOutputToken = swapDataCollateralToDebt;
+
+                    if (inputTokenName == "WETH") {
+                      outputToken = weth;
+                      await weth.deposit({ value: amountIn });
+                    }
+                  }
+
+                  subjectMinAmountOut = subjectSetAmount.div(2);
+                  subjectSetToken = setToken.address;
+                  await setToken.approve(flashMintLeveraged.address, subjectSetAmount);
+
+                  if (inputTokenName != "ETH") {
```

Comment:
```suggestion
                  if (inputTokenName !== "ETH") {
```

Reply:
resolved in:
https://github.com/IndexCoop/index-coop-smart-contracts/pull/143/commits/15b7fac84f9f9ffcdbc1ae0ca3c90bf27af95a36